### ISSUE TITLE
_nodes API prints hostname inconsistently

### ISSUE THREAD

**Issue by pickypg** [elastic/elasticsearch#17604]
ES 2.x / 5.0

APIs that depend on the printing of `NetworkAddress.format` for `TransportAddress` objects, such as `_nodes/transport` currently print out the local node versus non-local nodes (e.g., `n - 1` nodes) inconsistently.

If the published address involved a hostname, then the local node will have resolved its hostname and it will report its hostname alongside its IP address and port. However, since it does not know/care about the hostname of _most_ other nodes, then it will not have resolved their hostname and it will not print it out. For example: if you start two nodes with

```shell
$ bin/elasticsearch -Des.network.publish_host=`hostname` -Des.network.bind_host=127.0.0.1,`hostname`
```

Then try to find the transport address of those nodes:

```shell
$ curl -XGET localhost:9201/_nodes/transport?pretty
```

Then you'll get a response like (ignoring node uid and name, which are random):

```json
{
  "cluster_name": "elasticsearch",
  "nodes": {
    "cnPqK1eyQUykNZkhRXCeJQ": {
      "name": "Icemaster",
      "transport_address": "Chriss-MBP.home/192.168.1.6:9301",
      "host": "192.168.1.6",
      "ip": "192.168.1.6",
      "version": "2.3.0",
      "build": "8371be8",
      "http_address": "Chriss-MBP.home/192.168.1.6:9201",
      "transport": {
        "bound_address": [
          "127.0.0.1:9301",
          "[fe80::9425:ff:fe76:2429]:9301",
          "[fe80::82e6:50ff:fe0a:e71a]:9301",
          "192.168.1.6:9301"
        ],
        "publish_address": "Chriss-MBP.home/192.168.1.6:9301",
        "profiles": {}
      }
    },
    "_yVpRl1pSRa26EHBbr8ZHA": {
      "name": "Masque",
      "transport_address": "192.168.1.6:9300",
      "host": "192.168.1.6",
      "ip": "192.168.1.6",
      "version": "2.3.1",
      "build": "bd98092",
      "http_address": "Chriss-MBP.home/192.168.1.6:9200",
      "transport": {
        "bound_address": [
          "127.0.0.1:9300",
          "[fe80::9425:ff:fe76:2429]:9300",
          "[fe80::82e6:50ff:fe0a:e71a]:9300",
          "192.168.1.6:9300"
        ],
        "publish_address": "192.168.1.6:9300",
        "profiles": {}
      }
    }
  }
}
```

If you instead invoke the same API from the node running at port `9200`, then its `transport_address` will be reported properly with its hostname, while the other node will lose the hostname. The "problem" is that the `TransportAddress` address used by `n - 1` of the nodes is serialized and its hostname gets forgotten on other nodes, so when it gets printed, we avoid doing a hostname lookup and end up with inconsistent results.

The inconsistency can lead to parsing failure when unexpected and it leads to inconsistent handling/behavior. The solution is pretty straight forward: stop printing the hostname. This has the added benefit that it means that the format is always _directly_ usable within a URI without any parsing.

**Comment by pickypg** [comment]
Reopened for the #17606 version since it wasn't just a cherry-pick.

**Comment by pickypg** [comment]
Closed by #17606.
